src/common/hashtable/src/utils.rs

```diff
@@ -185,3 +185,117 @@ pub mod sse {
         }
     }
 }
+
+// for merge into:
+// we use BlockInfoIndex to maintain an index for the block info in chunks.
+pub struct BlockInfoIndex {
+    // the intervals will be like below:
+    // (0,10)(11,29),(30,38). it's ordered.
+    #[allow(dead_code)]
```

Comment:
`#[allow(dead_code)]` could move into top of BlockInfoIndex.

Reply:
This pr is not finished.